src/Stripe.net/Services/Sources/StripeSourceService.cs

```diff
@@ -40,6 +40,19 @@ public virtual StripeSource Detach(string customerId, string sourceId, StripeReq
             );
         }
 
+        public virtual StripeList<StripeSource> List(string customerId, StripeSourceListOptions listOptions = null, StripeRequestOptions requestOptions = null)
+        {
+            var url = string.Format(Urls.CustomerSources, customerId);
+
+            if (listOptions == null) {
+                listOptions = new StripeSourceListOptions();
+            }
```

Comment:
Why do you need to do this? Is it to ensure that `object` is passed as `source` even if nothing is sent? Asking because we don't do this in other List methods such as [customer](https://github.com/stripe/stripe-dotnet/blob/f157cede05b8e3ceb2dcac658fa5190078f117fc/src/Stripe.net/Services/Customers/StripeCustomerService.cs#L50-L56)

Reply:
Yes, exactly. The `StripeSourceListOptions` object will ensure that `object=source` is added as a query parameter in the request. This seemed less hacky than hardcoding the parameter in the URL, but it does force us to instantiate an object if none is provided.